charts/partials/templates/_proxy.tpl

```diff
@@ -119,8 +125,14 @@ lifecycle:
         - -c
         - sleep {{.Values.global.proxy.waitBeforeExitSeconds}}
 {{- end }}
-{{- if or (not .Values.global.proxy.disableIdentity) (.Values.global.proxy.saMountPath) }}
+{{- if or (not .Values.global.proxy.disableIdentity) (.Values.global.proxy.saMountPath) (.Values.global.proxy.trace) }}
 volumeMounts:
+{{- if (.Values.global.proxy.trace) }}
+{{- if (.Values.global.proxy.trace.collectorSvcAddr)}}
```

Comment:
We have to have separate checks because helm dosen't support short-cut evaluations. And checking for `.Values.global.proxy.trace.collectorSvcAddr` when `.Values.global.proxy.trace` is nil throws an error.

Reply:
Also, Haven't added the same volumes, etc for `controlPlaneTracing`, as I felt that these tags are net necessary for them, but I can do it if you think it is!

Reply:
Not having these tags in the control plane components would mean they wouldn't get a jaegger link in the dashboard (should we go down that route), right? I think they deserve the same treatment as non-control-plane pods :-)

Reply:
On the other hand, regarding these lines, I think we can assume that `Values.global.proxy.trace` is always set as per the `values.yaml` structure. So would only need to check on `Values.global.proxy.trace.collectorSvcAddr`. Also, `patch.json` is only checking for `Values.global.proxy.trace`; if I'm not mistaking we should change that to `Values.global.proxy.trace.collectorSvcAddr`.

Reply:
@alpeb Found the issue that was causing `Values.global.proxy.trace` to be nil both in install and inject paths, and updated it. :)